tests/test_constants.py

```diff
@@ -1,177 +1,168 @@
+    (CRS('3857'), "CRS('3857')"),
+    (CRS('4326'), "CRS('4326')"),
+    (CRS('32633'), "CRS('32633')"),
+    (CRS('32733'), "CRS('32733')"),
+])
+def test_crs_repr(crs, crs_repr):
+    assert crs_repr == repr(crs)
+
+
+@pytest.mark.parametrize('crs', CRS)
+def test_crs_has_value(crs):
+    assert CRS.has_value(crs.value), f'Expected support for CRS {crs.value}'
+
+
+@pytest.mark.parametrize('value, fails', [
+    ('string', True), (-1, True), (999, True), (None, True), (3035, False), ('EPSG:3035', False), (10000, False)
+
+])
+def test_custom_crs(value, fails):
+    if fails:
+        with pytest.raises(ValueError):
+            CRS(value)
+    else:
+        CRS(CRS(value))
+
+        new_enum_value = str(value).lower().strip('epsg: ')
+        assert CRS.has_value(new_enum_value)
+
+
+@pytest.mark.parametrize('crs', [CRS.WGS84, CRS.POP_WEB, CRS.UTM_38N])
+def test_pyproj_methods(crs):
+    assert isinstance(crs.projection(), pyproj.Proj)
+    assert isinstance(crs.pyproj_crs(), pyproj.CRS)
+
+
+@pytest.mark.parametrize('ext, mime_type', [
+    ('tif', MimeType.TIFF),
+    ('jpeg', MimeType.JPG),
+    ('jpg', MimeType.JPG),
+    ('h5', MimeType.HDF),
+    ('hdf5', MimeType.HDF),
+    *[(mime_type.extension, mime_type) for mime_type in MimeType]
+])
+def test_mime_type_from_string(ext, mime_type):
+    assert MimeType.from_string(ext) == mime_type
+
+
+@pytest.mark.parametrize('faulty_arg', ['unknown ext', 'tiff;depth=32f'])
+def test_mimetype_no_value_fail(faulty_arg):
+    assert not MimeType.has_value(faulty_arg), 'This value is not supposed to be type of the Enum'
+    with pytest.raises(ValueError):
+        MimeType.from_string(faulty_arg)
+
+
+@pytest.mark.parametrize('ext', ['tif', 'tiff', 'jpg', 'jpeg', 'png', 'jp2'])
+def test_is_image_format(ext):
+    mime_type = MimeType.from_string(ext)
+    assert MimeType.is_image_format(mime_type)
+
+
+@pytest.mark.parametrize('mime_type, expected_string', [
+    (MimeType.PNG, 'image/png'),
+    (MimeType.JPG, 'image/jpeg'),
+    (MimeType.TIFF, 'image/tiff'),
+    (MimeType.JSON, 'application/json'),
+    (MimeType.CSV, 'text/csv'),
+    (MimeType.ZIP, 'application/zip'),
+    (MimeType.HDF, 'application/x-hdf'),
+    (MimeType.XML, 'text/xml'),
+    (MimeType.TXT, 'text/plain'),
+    (MimeType.TAR, 'application/x-tar'),
+])
+def test_get_string(mime_type, expected_string):
+    assert MimeType.get_string(mime_type) == expected_string
+
+
+def test_get_expected_max_value():
+    assert MimeType.TIFF.get_expected_max_value() == 65535
+    assert MimeType.PNG.get_expected_max_value() == 255
+
+    with pytest.raises(ValueError):
+        MimeType.TAR.get_expected_max_value()
+
+
+def test_request_type():
+    with pytest.raises(ValueError):
+        RequestType('post')
+
+    with pytest.raises(ValueError):
+        RequestType('get')
+
+    try:
+        RequestType('POST')
+        RequestType('GET')
+    except BaseException:
+        assert False, 'Could not instantiate enum'
```

Comment:
You can actually remove try-catch statement here. If a call `RequestType('XYZ')` fails, pytest handles this nicely.